### PR TITLE
WFLY-19071/WFLY-19072: Fixing AddressSettings default values and Artemis 2.32 dependencies

### PR DESCRIPTION
**Pull request by ehsavoie**
* Setting the default values on the root of AddressSettings so that
    they are used by all child AddressSettings.
  * Fixing the JSON produced to be retro compatible.
  * Adding module de.dentrassi.crypto.pem-keystore
  * Moving from sunb.jdk to java.base   

Jira: https://issues.redhat.com/browse/WFLY-19072
      https://issues.redhat.com/browse/WFLY-19071

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]

File: testsuite/shared/src/main/java/org/jboss/as/test/shared/LayersTestBase.java [hunk 1]
```diff
@@ -252,8 +252,8 @@ public abstract class LayersTestBase {
             "javax.sql.api",
             "javax.xml.stream.api",
             // TODO see if org.apache.activemq.artemis.client can use specific JDK module(s)
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/shared/LayersTestBase.java:254]: Please remove the TODO comment.

File: ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/de/dentrassi/crypto/pem-keystore/main/module.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ Copyright The WildFly Authors
+  ~ SPDX-License-Identifier: Apache-2.0
+  -->
+
+<module xmlns="urn:jboss:module:1.9" name="de.dentrassi.crypto.pem-keystore">
```
Comment (bstansberry) [inline ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/de/dentrassi/crypto/pem-keystore/main/module.xml:8]: @fjuma FYI about this new module.

**Review by bstansberry — CHANGES_REQUESTED** [review]
@ehsavoie Just one minor change, plus I wanted to make Farah (and @darranl) aware of this new crypto lib.

**Review by bstansberry — APPROVED** [review]

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ManagementUtil.java [hunk 3]
```diff
@@ -28,6 +33,11 @@
  */
 public class ManagementUtil {
 
+    private static final String SNAKE_REPLACEMENT = "$1-$2";
+    private static final String CAMEL_REGEXP = "([a-z])([A-Z]+)";
+    private static final Map<String, AttributeDefinition> ADDESS_SETTINGS_ATTRIBUTES =
```
Comment (rhusar) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ManagementUtil.java:38]: This should be `ADDRESS_SETTINGS_ATTRIBUTES`

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ManagementUtil.java [hunk 4]
```diff
@@ -28,6 +33,11 @@
...
+            Arrays.stream(AddressSettingDefinition.ATTRIBUTES).collect(Collectors.toMap(AttributeDefinition::getName, att -> att)) ;
```
Comment (rhusar) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ManagementUtil.java:39]: Replace with `Function.identity()`

**Review by rhusar — CHANGES_REQUESTED** [review]
Also, `ADDESS_SETTINGS_ATTRIBUTES` is never used, what's up with that? :)

**Review by rhusar — APPROVED** [review]
LGTM now, feedback has been addressed.

**Comment by bstansberry** [comment]
Thanks @ehsavoie and @rhusar

**Pull request merged**